Assets/Features/Building/Scripts/Grid/Grid.cs

```diff
@@ -0,0 +1,391 @@
+                    {
+                        _cells[x, y, z] = CellData.empty;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// If the map update was called this frame update it
+        /// </summary>
+        private void LateUpdate()
+        {
+            if (_mapUpdated)
+            {
+                UpdateMap();
+                _mapUpdated = false;
+            }
+        }
+
+        /// <summary>
+        /// Loops through the whole grid and sets all the cells to what they are supposed to be
+        /// TODO add a buffer, so correct tiles aren't changed
+        /// </summary>
+        private void UpdateMap()
+        {
+            for (int x = 0; x < _gridSize.x; x++)
+            {
+                for (int y = 0; y < _gridSize.y; y++)
+                {
+                    for (int z = 0; z < _gridSize.z; z++)
+                    {
+                        var cell = _cells[x, y, z];
+
+                        Vector3Int offset = new Vector3Int(Mathf.RoundToInt(_tilemap.transform.position.x),
+                            Mathf.RoundToInt(_tilemap.transform.position.y),
+                            Mathf.RoundToInt(_tilemap.transform.position.z));
+
+
+                        TileChangeData tile = new TileChangeData()
+                        {
+                            position = new Vector3Int(x, y, z) - offset,
+                        };
+
+                        if (cell.Tile != CellData.empty.Tile)
+                        {
+                            tile.transform = Matrix4x4.Rotate(Quaternion.Euler(0, 0, cell.Rotation * -90));
+                            tile.tile = _atlas.Items[cell.Tile].Tile;
+
+                            float buildAmount = cell.BuildPercentage;
+                            tile.color = new Color(1, 1, 1, buildAmount);
+                        }
+
+                        _tilemap.SetTile(tile, true);
+                    }
+                }
+            }
+        }
+
+        public bool BuildTile(Vector3Int gridVector, float speed)
+        {
+            if (Get(gridVector) == -1)
+                return true;
+        
+            var buildTiles = new List<Vector3Int>() { gridVector };
+            for (int i = 0; i < _cellGroup.Count; i++)
+            {
+                if (_cellGroup[i].Contains(gridVector))
+                {
+                    foreach (var child in _cellGroup[i])
+                    {
+                        if (child != buildTiles[0])
+                            buildTiles.Add(child);
+                    }
+                }
+            }
+
+            bool isFinished = false;
+            foreach (var tile in buildTiles)
+            {
+                isFinished = _cells[tile.x, tile.y, tile.z].Build(speed * Time.deltaTime);
+            }
+
+            _mapUpdated = true;
+
+            return isFinished;
+        }
+
+        /// <summary>
```

Comment:
Be consistent with documentation, either do everything or nothing. Rather have the first one though